### PR TITLE
blcloud seedui initial commit

### PR DESCRIPTION
**Pull request by vigneshranganathan-eaton**
#### Changes proposed in this Pull Request:

-  New feature SeedUI Template which contains Admin invite, Admin user invite registration process, Self invite, Self invite registration process and Login. 

#### Screenshots / Screen Recording (if applicable)

-
image

#### To Test:
         Follow the steps given in below confluence link to validate SeedUI

- https://confluence-prod.tcc.etn.com/display/IoTPlatform/BLUI+-+CLI+-+Installation+Steps

### CODE REVIEW

**Comment by JeffGreiner-eaton** [comment]
@vigneshranganathan-eaton 
I need assistance with some final configuration items for the cli seed ui.

image

npm install @etn-electrical/derms-blcloud-react-seedui returns 401 Unauthorized and need the valid criteria for adopterId, adopterApplicationName and applicationId.

You can ping me in teams or email
thank you

**Comment by JeffGreiner-eaton** [comment]
Just some high level feedback on a few items (and to get this published)
I see a warning that seems to be from [derms-blcloud-react-seedui](https://github.com/orgs/etn-electrical/packages/npm/package/derms-blcloud-react-seedui) package and throws warning at login and navigation.

image

Also the drawer needs some tweaks when page is resized or is mobile view. (overall the app needs some view ports set)
Might try something like this for drawer >

```variant={isMobile ? 'temporary' : 'persistent'}``` and you can refer back to the routing template for this if you want to update everything to be more responsive & be mobile ready.

The invite and register emails do not currently work and not sure if it was intended to work but the links are these and don't load.

```https://localhost:3000/inviteregistration?invitationkey=f915f195925b45fa88d82c23cdfdd743&isExistingUser=False&expiryTime=1689872088```
and
```https://localhost:3000/selfregistration?invitationkey=ec6d7c6d748546c7af76fd7b92a4007a&isExistingUser=False&expiryTime=1689879394```

The self-registration success screen could use a full width button labeled "Okay" or something that returns back to login screen.

image

**Comment by vigneshranganathan-eaton** [comment]
> Just some high level feedback on a few items (and to get this published) I see a warning that seems to be from [derms-blcloud-react-seedui](https://github.com/orgs/etn-electrical/packages/npm/package/derms-blcloud-react-seedui) package and throws warning at login and navigation.
> 
> image
> 
> Also the drawer needs some tweaks when page is resized or is mobile view. (overall the app needs some view ports set) Might try something like this for drawer >
> 
> `variant={isMobile ? 'temporary' : 'persistent'}` and you can refer back to the routing template for this if you want to update everything to be more responsive & be mobile ready.
> 
> The invite and register emails do not currently work and not sure if it was intended to work but the links are these and don't load.
> 
> `https://localhost:3000/inviteregistration?invitationkey=f915f195925b45fa88d82c23cdfdd743&isExistingUser=False&expiryTime=1689872088` and `https://localhost:3000/selfregistration?invitationkey=ec6d7c6d748546c7af76fd7b92a4007a&isExistingUser=False&expiryTime=1689879394`
> 
> The self-registration success screen could use a full width button labeled "Okay" or something that returns back to login screen.
> 
> image

**Comment by vigneshranganathan-eaton** [comment]
> 

@JeffGreiner-eaton I have taken care the warnings and other issues. Then registration mail link is "https" so it will work in env. So now we have made it to "http" for local. you can validate now.

**Comment by JeffGreiner-eaton** [comment]
> > 
> 
> @JeffGreiner-eaton I have taken care the warnings and other issues. Then registration mail link is "https" so it will work only in env. So now we have made it to "http" for local. you can validate now.

Since the update I'm receiving 500 internal server error on invite & register

image
image

**Comment by JeffGreiner-eaton** [comment]
Not sure what the plans are for deep linking but can deep link to /invite & /dashboard without logging in.

**Comment by JeffGreiner-eaton** [comment]
The static build main is slightly large, might need to do code split or have a plan on some additional documentation for end user.

File sizes after gzip:
 
  ```3.51 MB  build/static/js/main.aa71eb88.js```
  2.52 kB  build/static/css/main.b8b2b90b.css
 
The bundle size is significantly larger than recommended.
Consider reducing it with code splitting: https://goo.gl/9VhYWB
You can also analyze the project dependencies: https://goo.gl/LeUzfb

**Comment by JeffGreiner-eaton** [comment]
the @etn-electrical/derms-blcloud-react-seedui package needs some version updates

- @pxblue/prettier-config@1.0.3: This package has been deprecated. Please update to the new @brightlayer-ui/prettier-config. 
-  @mui/styles@5.14.1" has incorrect peer dependency "react@^17.0.0".

File: seedui-typescript/template/src/component/layout/PageLayout.tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    useEffect(() => {
+        const activeRoute = ROUTE_OBJ.find((item) => pathname.indexOf(item.path) !== -1);
+        setActive((activeRoute && activeRoute.name) || 'inviteUser');
+    }, [pathname]);
+
+    return (
+        <DrawerLayout
+            sx={{ height: '100%' }}
+            drawer={
+                <Drawer open={drawerOpen} width={332} variant={isMobile ? 'temporary' : 'persistent'} activeItem={active}>
```
Comment (JeffGreiner-eaton) [inline seedui-typescript/template/src/component/layout/PageLayout.tsx:84]: you can add set drawer open(false) for backDrop click..

ModalProps={{
                        onBackdropClick: (): void => {
                            setDrawerOpen(false);
                        },
                    }}

File: seedui-typescript/template/src/component/layout/PageLayout.tsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+                window.location.href = '/dashboard';
+            },
+            divider: false,
+        },
+        {
+            itemID: 'inviteUser',
+            title: 'Invite User',
+            icon: <GroupAdd />,
+            onClick: (): void => {
+                setActive('inviteUser');
```
Comment (JeffGreiner-eaton) [inline seedui-typescript/template/src/component/layout/PageLayout.tsx:58]: can you add something here to setDrawerOpen(false) for onclick so when nav item is selected the drawer closes and page is displayed...

onClick:
                                page.route !== undefined
                                    ? (): void => {
                                          handleNavigate(page.route);
                                          if (isMobile) setDrawerOpen(false);
                                      }

Comment (JeffGreiner-eaton) [reply seedui-typescript/template/src/component/layout/PageLayout.tsx:58]: also something like this for handleNav

const handleNavigate = useCallback(
        (id: string): void => {
            navigate(id);
            setSelected(id);
        },
        [navigate, setSelected]
    );

File: seedui-typescript/template/src/component/layout/PageLayout.tsx (hunk 1 above)
Comment (vigneshranganathan-eaton) [reply seedui-typescript/template/src/component/layout/PageLayout.tsx]: done with the changes

**Comment by vigneshranganathan-eaton** [comment]
@JeffGreiner-eaton I am done with required changes kindly review it.

**Comment by JeffGreiner-eaton** [comment]
> @JeffGreiner-eaton I am done with required changes kindly review it.

please let me know when the registration services are running so I can verify entire template

**Review by surajeaton — COMMENTED** [review]
Please resolve the conflicts.

**Comment by vigneshranganathan-eaton** [comment]
> Please resolve the conflicts.

@surajeaton Conflicts resolved

**Comment by vigneshranganathan-eaton** [comment]
> > @JeffGreiner-eaton I am done with required changes kindly review it.
> 
> please let me know when the registration services are running so I can verify entire template

@JeffGreiner-eaton  I have updated with new Admin invite and registration API is up and running, you can validate the functionality now.

**Comment by JeffGreiner-eaton** [comment]
1. Noticed the derms package name changed from @etn-electrical/derms-blcloud-react-seedui to @etn-electrical/derms-blcloud-seedui and need to update UserInvite.tsx, App.tsx and AuthUIActions.tsx imports.

2. Error on failed prop at startup
image

3. Error on unique key at startup
image

4. RegistrationComplete warning
image
image

5. Setup org details finish button enabled (need required fields)
image

6. Setup org details page MUI warnings for each input
image

7. Setup org details page character limits Address, Address line 2 and City (currently only 15 char)
image

8. Registration link expires too quickly or set as random (remove @eaton email and use something else) support@support or something
image

9. Success screens displays last name then first. Not a huge deal but could be reversed
image

**Comment by JeffGreiner-eaton** [comment]
Note - I did not verify any new features related to user management, roles or multiple invites due to not being available in the updated derms package

**Comment by joebochill** [comment]
Closing this as we're working to innersource the DERMS workflow code into the blui community repository instead. We will revisit this after.

**Pull request closed without merging**